### PR TITLE
Update RandomUtils and SingleParticleGenerator to the latest changes in the petalo branch

### PR DESCRIPTION
**Pull request by paolafer**
This PR adds to the nexus core the latest changes carried out in the `petalo` branch regarding the random generation of momentum and energy of an individual particle. The code itself has already passed the review, it only needs to be checked that the code compiles and runs (in the case mistakes had been introduced in the copy), which is done automatically.

### CODE REVIEW

File: source/generators/SingleParticleGenerator.cc [hunk 1]
```diff
@@ -62,11 +61,11 @@ costheta_max_(1.), phi_min_(0.), phi_max_(2.*pi)
   msg_->DeclareProperty("region", region_,
     "Set the region of the geometry where the vertex will be generated.");
 
-  msg_->DeclareProperty("momentum_X", momentum_X_,
+  msg_->DeclareProperty("momentum_X", momentum_[0],
```
Comment (jmalbos) [inline source/generators/SingleParticleGenerator.cc]: We could simplify this using the function `DeclarePropertyWithUnit`, which accepts 3-vectors (see https://geant4.kek.jp/lxr/source/intercoms/src/G4GenericMessenger.cc#L106).

File: source/utils/RandomUtils.cc [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  {
+    if (max_value == min_value){
+      return min_value;
+    } else {
+      return (G4UniformRand()*(max_value - min_value) + min_value);
+    }
+  }
+
+
+  G4ThreeVector Direction(G4double costheta_min, G4double costheta_max,
```
Comment (jmalbos) [inline source/utils/RandomUtils.cc]: `Direction` is not a very descriptive name. Why not `UniformRandomDirection` or something similar?

Comment (paolafer) [reply source/utils/RandomUtils.cc]: What about `RandomDirection`?

Comment (paolafer) [reply source/utils/RandomUtils.cc]: Actually, it should be `RandomDirectionInRange`.

Comment (jmalbos) [reply source/utils/RandomUtils.cc]: Yes, that's fine too.

**Comment by jmalbos** [comment]
This PR updates the `SingleParticleGenerator` cleaning up the code and making use of new random-generation functions (added to `utils`). The new code compiles and runs. I approve it.

**Pull request merged**